src/weights.jl

```diff
@@ -14,6 +14,7 @@ macro weights(name)
             sum::S
         end
         $(esc(name))(values::AbstractVector{<:Real}) = $(esc(name))(values, sum(values))
+        $(esc(:weightstype))(::Type{<:$(esc(name))}) = $(esc(name))
```

Comment:
There's probably a simpler way of doing this but I couldn't figure it out...

Reply:
out of curiosity - why do we need this macro? I usually try to avoid such things as they make debugging later harder.

Reply:
The macro just avoids some duplication as we define 4 almost identical types. We could equally well use `@eval` in a loop.

Reply:
I know. I just say we save some 10 lines of code at the expense of making it less clear for normal users. Of course it can stay from a technical standpoint.